src/lib/file-extractor.ts

```diff
@@ -63,6 +64,11 @@ export function convertPdfToImages(pdfPath: string, outputPrefix: string): strin
     const data = readFileSync(fullPath);
     // Optionally prefix with "data:image/png;base64," for consistency
     const base64Encoded = `data:image/png;base64,${data.toString('base64')}`;
+    try {
+      unlinkSync(fullPath); // remove temp file
+    } catch (error) {
+      console.error(`Failed to remove temporary file: ${fullPath}`, error);
```

Comment:
@pkarw _**Question**_: Should there be error handling in case the base64 encoding fails for whatever reason?